debug/video-export.html

```diff
@@ -0,0 +1,111 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Mapbox GL JS debug page</title>
+    <meta charset='utf-8'>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
+    <link rel='stylesheet' href='../dist/mapbox-gl.css' />
+    <style>
+        #map { width: 960px; height: 540px; }
+    </style>
+</head>
+
+<body>
+<div id='map'></div>
+
+<script src='../dist/mapbox-gl-dev.js'></script>
+<script src='../debug/access_token_generated.js'></script>
+
+<script type="module">
+import loadEncoder from 'https://unpkg.com/mp4-h264@1.0.6/build/mp4-encoder.js';
+import {simd} from "https://unpkg.com/wasm-feature-detect?module";
+
+const map = window.map = new mapboxgl.Map({
+    container: 'map',
+    center: [7.533634776071096, 45.486077107185565],
+    zoom: 13.5,
+    pitch: 61,
+    bearing: -160,
+    style: 'mapbox://styles/mapbox/satellite-v9'
+});
+
+async function animate() {
+    // do all the animations you need to record here
+    map.easeTo({
+        bearing: map.getBearing() - 20,
+        duration: 3000,
+        easing: t => t
+    });
+    // wait for animation to finish
+    await untilMapEvent('moveend');
+}
+
+map.on('load', async () => {
+    map.addSource('dem', {type: 'raster-dem', url: 'mapbox://mapbox.mapbox-terrain-dem-v1'});
+    map.setTerrain({source: 'dem', exaggeration: 1.5});
+
+    // wait until the map settles
+    await untilMapEvent('idle');
+
+    // uncomment to fine-tune animation without recording:
+    // animate(); return;
+
+    // don't forget to enable WebAssembly SIMD in chrome://flags for faster encoding
+    const supportsSIMD = await simd();
+
+    // initialize H264 video encoder
+    const Encoder = await loadEncoder({simd: supportsSIMD});
+
+    const gl = map.painter.context.gl;
+    const width = gl.drawingBufferWidth;
+    const height = gl.drawingBufferHeight;
+
+    const encoder = Encoder.create({
+        width,
+        height,
+        fps: 60,
+        quantizationParameter: 20,
+        rgbFlipY: true
+    });
+
+    // stub performance.now for deterministic rendering per-frame (only available in dev build)
+    let now = performance.now();
+    mapboxgl.setNow(now);
```

Comment:
This is only available with dev builds, I believe. This example isn't directly usable with the built version of GL JS on CDN or NPM. That's fine for this debug page, but it's good to note.

Reply:
@ryanhamley yes, I mentioned this in the PR description and the comment above.

Reply:
Added another commit that documents & exposes these methods in production build, so we can add the official example after this gets into a release.

Reply:
Ooops I missed that in the PR description.

> Note that it no longer falls back to Date.now because it should be universally supported now.

👍 awesome